specs/core/0_beacon-chain.md

```diff
@@ -1134,32 +1135,32 @@ Before genesis has been triggered and whenever the deposit contract emits a `Dep
 * `deposits` is the list of all deposits, ordered chronologically, up to and including the deposit triggering the latest `Deposit` log
 * `timestamp` is the Unix timestamp in the Ethereum 1.0 block that emitted the latest `Deposit` log
 
-When `is_genesis_trigger(deposits, timestamp) is True` for the first time let:
+When `is_genesis_trigger(deposits, timestamp, deposit_root) is True` for the first time let:
 
 * `genesis_deposits = deposits`
 * `genesis_time = timestamp - timestamp % SECONDS_PER_DAY + 2 * SECONDS_PER_DAY` where `SECONDS_PER_DAY = 86400`
+* `deposit_root` is the Merkle tree root of the data of the given `deposits`
 * `genesis_eth1_data` be the object of type `Eth1Data` where:
-    * `genesis_eth1_data.block_hash` is the Ethereum 1.0 block hash that emitted the log for the last deposit in `deposits`
     * `genesis_eth1_data.deposit_root` is the deposit root for the last deposit in `deposits`
     * `genesis_eth1_data.deposit_count = len(genesis_deposits)`
+    * `genesis_eth1_data.block_hash` is the Ethereum 1.0 block hash that emitted the log for the last deposit in `deposits`
 
 *Note*: The function `is_genesis_trigger` has yet to be agreed by the community, and can be updated as necessary. We define the following testing placeholder:
 
 ```python
-def is_genesis_trigger(deposits: List[Deposit], timestamp: uint64) -> bool:
+def is_genesis_trigger(deposits: List[Deposit], timestamp: uint64, deposit_root: Hash) -> bool:
```

Comment:
There's no need to add `deposit_root`. It's redundant information (can be derived from `deposits`).

Reply:
@JustinDrake to confirm, that means we have add [prepare_genesis_deposits](https://github.com/ethereum/eth2.0-specs/blob/6aef6c5634acc0c378c1f03b2cefd32ba3501954/test_libs/pyspec/eth2spec/test/helpers/deposits.py#L63-L90) logic to `is_genesis_trigger`, including calling [`calc_merkle_tree_from_leaves`, `get_merkle_root`, and `get_merkle_proof` (in `merkle_minimal.py`)](https://github.com/ethereum/eth2.0-specs/blob/e8b4c4c57f1f07a8eae6c88249bad6c19335017f/test_libs/pyspec/eth2spec/utils/merkle_minimal.py#L12-L24).

I think it's good to move the `merkle_minimal.py` to the spec, but the `zerohashes` trick might not fit the spec format well...

Reply:
I'm happy to rework the PR :) I think we can just roll back to what it was, and add the one line that initialises `deposit_root` to the correct thing (as opposed to all zeros).